openedx_lti_tool_plugin/resource_link_launch/views.py

```diff
@@ -0,0 +1,409 @@
+            # Enroll user.
+            self.enroll(request, edx_user, CourseKey.from_string(course_id))
+            # Get launch response.
+            resource_response, context_key = self.get_launch_response(
+                request,
+                edx_user,
+                course_id,
+                usage_key_string,
+            )
+            # Handle AGS.
+            self.handle_ags(
+                launch_message,
+                launch_data,
+                lti_profile,
+                context_key,
+            )
+
+            return resource_response
+        except LtiToolLaunchException as exc:
+            return LoggedHttpResponseBadRequest(_(f'LTI 1.3 Launch failed: {exc}'))
+
+    def get_launch_data(self, launch_message: DjangoMessageLaunch) -> dict:
+        """Get launch data from launch message.
+
+        Args:
+            launch_message: DjangoMessageLaunch object.
+
+        Returns:
+            Dictionary with launch message data.
+
+        Raises:
+            LtiToolLaunchException: If LtiException is raised.
+
+        """
+        try:
+            return launch_message.get_launch_data()
+        except LtiException as exc:
+            raise LtiToolLaunchException(
+                _(f'Launch message validation failed: {exc}'),
+            ) from exc
+
+    def get_identity_claims(
+        self,
+        launch_data: dict
+    ) -> Tuple[str, str, str, dict]:
+        """Get identity claims from launch data.
+
+        Args:
+            launch_data: Launch data dictionary.
+
+        Returns:
+            A tuple containing the `iss`, `aud`, `sub` and PII claims.
+
+        .. _OpenID Connect Core 1.0 - ID Token:
+            https://openid.net/specs/openid-connect-core-1_0.html#IDToken
+
+        .. _OpenID Connect Core 1.0 - Standard Claims:
+            https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims
+
+        """
+        return (
+            launch_data.get('iss'),
+            get_client_id(launch_data.get('aud'), launch_data.get('azp')),
+            launch_data.get('sub'),
+            get_pii_from_claims(launch_data) if SAVE_PII_DATA.is_enabled() else {},
+        )
+
+    def check_course_access_permission(self, course_id: str, iss: str, aud: str):
+        """Check course access permission.
+
+        This function will check if the given `course_id` is allowed by the
+        CourseAccessConfiguration instance of this launch `LtiTool`.
+
+        Args:
+            course_id: Course ID string.
+            iss: Issuer claim.
+            aud: Audience claim.
+
+        Raises:
+            LtiToolLaunchException: If CourseAccessConfiguration instance
+                does not exist for LtiTool or the `course_id` is not allowed.
+
+        """
+        if not COURSE_ACCESS_CONFIGURATION.is_enabled():
+            return
+
+        lti_tool_config = self.tool_config.get_lti_tool(iss, aud)
+        course_access_config = CourseAccessConfiguration.objects.filter(
```

Comment:
@anfbermudezme I think it could be better but I prefer not to address this on this PR since it will require adding more changes that aren't a priority for this PR